File: algorithms/0011.container-with-most-water/solution.go
func maxArea(height []int) int {
	max := 0
	left, right := 0, len(height)-1

	for left < right {
		amount := min(height[left], height[right]) * (right - left)

		if max < amount {
			max = amount
		}

		if height[left] > height[right] {
			right--
		} else {
			left++
		}
	}

	return max
}

func min(i int, j int) int {
	if i < j {
		return i
	}
	return j
}

File: algorithms/0011.container-with-most-water/solution.rb
# @param {Integer[]} height
# @return {Integer}
def max_area(height)
  max = 0
  left, right = 0, height.size - 1

  while left < right
    amount = [height[left], height[right]].min * (right - left)
    if amount > max
      max = amount
    end

    if height[left] > height[right]
      right -= 1
    else
      left += 1
    end
  end

  max
end
